src/components/Editor/SearchBar.js

```diff
@@ -155,10 +157,17 @@ class SearchBar extends Component<Props, State> {
     }
 
     if (this.props.searchOn && editor) {
-      const selection = editor.codeMirror.getSelection();
-      this.setState({ query: selection });
-      if (selection !== "") {
-        this.doSearch(selection);
+      const query = editor.codeMirror.getSelection() || this.state.query;
+
+      if (this.$input) {
+        this.$input.setFocus();
```

Comment:
An idea here to avoid the inner ref would be making a `focus` state field and having a prop to searchInput like `isFocused` that you assign it to.